ambry-frontend/src/main/java/com.github.ambry.frontend/FrontendMetrics.java

```diff
@@ -392,4 +384,50 @@ public FrontendMetrics(MetricRegistry metricRegistry) {
     putWithAccountAndContainerHeaderRate = metricRegistry.meter(
         MetricRegistry.name(AmbryBlobStorageService.class, "PutWithAccountAndContainerHeaderRate"));
   }
+
+  /**
+   * Class to hold different {@link RestRequestMetrics} for plain text, ssl, with and without encryption
+   */
+  class RestRequestMetricsGroup {
+    private final RestRequestMetrics blobMetrics;
+    private final RestRequestMetrics blobSSLMetrics;
+    private final RestRequestMetrics encryptedBlobMetrics;
+    private final RestRequestMetrics encryptedBlobSSLMetrics;
+
+    /**
+     * Instantiates {@link RestRequestMetricsGroup} for the given requestType and resource
+     * @param requestType refers to the type of request (Head/Get/Post)
+     * @param resource refers to the resource type (Blob/BlobInfo/UserMetadata)
+     */
+    RestRequestMetricsGroup(String requestType, String resource) {
+      blobMetrics = new RestRequestMetrics(AmbryBlobStorageService.class, requestType + resource, metricRegistry);
+      blobSSLMetrics =
+          new RestRequestMetrics(AmbryBlobStorageService.class, requestType + resource + SSL_SUFFIX, metricRegistry);
+      encryptedBlobMetrics =
+          new RestRequestMetrics(AmbryBlobStorageService.class, requestType + ENCRYPTED + resource, metricRegistry);
```

Comment:
Would you rather have the Encrypted in the middle of the name, or at the end? Depends on how you think you will tend to search for these metrics

Reply:
Moved encryption towards the end.